aws_xray_sdk/ext/pymongo/patch.py

```diff
@@ -0,0 +1,36 @@
+from pymongo import monitoring
+from aws_xray_sdk.core import xray_recorder
+
+
+class XrayCommandListener(monitoring.CommandListener):
+    """
+    A listener that traces all pymongo db commands to AWS Xray.
+    Creates a subsegment for each mongo db conmmand.
+
+    name: 'MongoDB-127.0.0.1:27017'
+    annotations: command_name:str, database_name: str, succeed: bool
+    """
+
+    def started(self, event):
+        host, port = event.connection_id
+        subsegment = xray_recorder.begin_subsegment(
+            f'MongoDB-{host}:{port}', 'remote')
+        subsegment.put_annotation('command_name', event.command_name)
+        subsegment.put_annotation('database_name', event.database_name)
```

Comment:
Any reason you put database name apart from the subsegment name? The common practice for current database clients is to use `database_name@database_url` as the subsegment name since it provides proper level of aggregation. You can see an example here: https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-sql. Please let me know if you have any particular reason to only name based on the host url.

Reply:
A `remote` subsegment will appear as a vertex in the X-Ray graph. I would imagine each vertex as one (logical) machine, or one service. If we use `database_name@database_url`, multiple databases in one MongoDB service will present as multiple vertexes, which may cause confusion. A subsegment naming `MongoDB-192.168.1.100:27017` will aggregate all requests to the same service, which is more convincing to me.

Reply:
For SQL database capturing all of our SDKs have been using naming scheme `name@url` and it works well so far. I'm not sure about the MondoDB use case but my concern is that for certain database that is slow, aggregating at host level would mask real issue or making debugging difficult as you need to drill down further to figure out if the issue is on the service level or database level.

Having said that, this is just something you might want to consider based on other db tracing we've been providing for a long time. Since this is the first PR support on MongoDB using host name might be a better practice for most MongoDB users. Please let me know your thoughts.